### PR TITLE
Add additional parameters for WS-Fed URL

### PR DESCRIPTION
**Pull request by joshcanhelp**
Add `options` parameter to `wsfed_url` method + tests

### CODE REVIEW

File: lib/auth0/api/authentication_endpoints.rb [hunk 1]
```diff
@@ -215,12 +215,22 @@ def samlp_url(connection = UP_AUTH)
...
         request_params = {
-          whr: connection
+          whr: connection,
+          wtrealm: options[:wtrealm],
+          wctx: options[:wctx],
+          wreply: options[:wreply]
         }
-        URI::HTTPS.build(host: @domain, path: "/wsfed/#{@client_id}", query: to_query(request_params))
+
+        url_client_id = @client_id if !request_params[:wtrealm]
```
Comment (lbalmaceda) [inline lib/auth0/api/authentication_endpoints.rb]: so `wtrealm` and `client_id` are not compatible?

File: lib/auth0/api/authentication_endpoints.rb [hunk 2]
```diff
@@ -215,12 +215,22 @@ def samlp_url(connection = UP_AUTH)
...
       # @see https://auth0.com/docs/api/authentication#accept-request35
       # @param connection [string] Connection to use; empty to show all
+      # @param options [hash] Extra options; supports wtrealm, wctx, wreply
       # @return [url] WS-Federation URL
-      def wsfed_url(connection = UP_AUTH)
+      def wsfed_url(connection = UP_AUTH, options = {})
         request_params = {
-          whr: connection
+          whr: connection,
+          wtrealm: options[:wtrealm],
```
Comment (lbalmaceda) [inline lib/auth0/api/authentication_endpoints.rb]: Please add a test that asserts these properties are not being sent in the body when "null" or missing

File: lib/auth0/api/authentication_endpoints.rb (hunk 1 above)
Comment (joshcanhelp) [reply lib/auth0/api/authentication_endpoints.rb]: https://auth0.com/docs/api/authentication#accept-request35

File: lib/auth0/api/authentication_endpoints.rb (hunk 2 above)
Comment (joshcanhelp) [reply lib/auth0/api/authentication_endpoints.rb]: This just generates a URL, no HTTP requests
